src/main.rs

```diff
@@ -22,17 +22,18 @@ pub fn get_test_names(config: &FlakeConfig) -> Result<Vec<String>, std::io::Erro
 pub fn run_single_test(setup: TestSetup) -> Result<TestResult, std::io::Error> {
     let mut result = TestResult::new(setup.name);
     for _ in 0..(setup.iterations as usize) {
-        let status = Command::new("sh")
+        let output = Command::new("sh")
             .arg("-c")
-            .stdout(Stdio::null())
-            .stderr(Stdio::null())
             .arg(&setup.command)
-            .status()
+            .output()
             .unwrap();
         result.iterations += 1;
-        if status.success() {
+        if output.status.success() {
             result.successes += 1;
         } else {
+            result
+                .failure_output
+                .push(String::from_utf8(output.stdout).unwrap());
```

Comment:
Let's store the `Vec<u8>` directly and try to coerce to `String` when we print failures. It's possible that the underlying byte vec won't be valid utf-8. If we store the byte vec directly we can attempt the `String` conversion and if that fails we can print the whole mess as hex or some such.